EIPS/eip-5805.md

```diff
@@ -0,0 +1,401 @@
+
+This function is thus optional, and its absence should be considered as a marker of the contract operating using block number. (This makes this EIP compatible with pre-existing voting contracts)
+
+```yaml
+- name: now
+  type: function
+  stateMutability: view
+  inputs: []
+  outputs:
+    - name: timepoint
+      type: uint256
+```
+
+#### getVotes
+
+This function returns the current voting weight of an account. This corresponds to all the voting power delegated to it at the moment this function is called.
+
+As tokens delegated to `address(0)` should not be counted/snapshoted, `getVotes(0)` SHOULD always return `0`.
+
+This function MUST be implemented
+
+```yaml
+- name: getVotes
+  type: function
+  stateMutability: view
+  inputs:
+    - name: account
+      type: address
+  outputs:
+    - name: votingWeight
+      type: uint256
+```
+
+#### getPastVotes
+
+This function returns the historical voting weight of an account. This corresponds to all the voting power delegated to it at a specific timepoint. The timepoint parameter should match the operating mode of the contract. This function SHOULD only serve past checkpoints that are immutable. Calling this function with a timepoint that is greater or equal to `now()` SHOULD revert. For any timepoint that is strictly smaller than `now()`, the value returned by `getPastVotes` should be constant.
+
+As tokens delegated to `address(0)` should not be counted/snapshoted, `getPastVotes(0,x)` SHOULD always return `0` (for all values of `x`).
+
+This function MUST be implemented
+
+```yaml
+- name: getPastVotes
+  type: function
+  stateMutability: view
+  inputs:
+    - name: account
+      type: address
+    - name: timepoint
+      type: uint256
+  outputs:
+    - name: votingWeight
+      type: uint256
+```
+
+#### delegates
+
+This function returns the address to which the voting power of an account is currently delegated.
+
+Note that if the delegate is `address(0)` then the voting power SHOULD NOT be checkpointed, and it should not be possible to vote with it.
+
+This function MUST be implemented
+
+```yaml
+- name: delegates
+  type: function
+  stateMutability: view
+  inputs:
+    - name: account
+      type: address
+  outputs:
+    - name: delegatee
+      type: address
+```
+
+#### delegate
+
+This function changes the caller's delegate, updating the vote delegation in the meantime.
+
+This function MUST be implemented
+
+```yaml
+- name: delegate
+  type: function
+  stateMutability: nonpayable
+  inputs:
+    - name: delegatee
+      type: address
+  outputs: []
+```
+
+#### delegateBySig
+
+This function changes an account's delegate using a signature, updating the vote delegation in the meantime.
+
+This function MUST be implemented
+
+```yaml
+- name: delegateBySig
+  type: function
+  stateMutability: nonpayable
+  inputs:
+    - name: delegatee
+      type: address
+    - name: nonce
+      type: uint256
+    - name: expiry
+      type: uint256
+    - name: v
```

Comment:
> [peer review] The `v` here could either be a boolean for the Y-parity, or consider add chainId to avoid replay attack. (note EIP-2294 is in the proposal to formalize the boundary of chainId to support a clear decision of this)

This is an interresting discussion!

My approach has been to match what currently exists (in particular EIP-2612), and make this EIP backward compatible with existing deployment (that afaik all reused EIP-2612 format).

I'll have a look at 5453

Reply:
That makes sense! I think I am going to proposal soon in Shanghai fork for the developers to make a decision either the chainId will be a full 256bit space for a likely less than 64bit space. Once that's done, I think some part of the execution layer needs to be updated. If the core developers want to allow a full 256bit space, then we will need to update how the parity bit is calculated